MyLibs/talk2screen.cpp

```diff
@@ -251,52 +308,17 @@ bool ScreenData::init_glad()
 	return true;
 }
 
-/* Load one texture for the entire screen */
-bool ScreenData::loadTextureIntoBuffers(std::vector<const char*> imgName, int texIdx)
-{
-	std::vector<unsigned char*> data;
-	for (int i = 0; i < texIdx; i++)
-	{
-		glGenTextures(1, &texture0[i]);
-		glBindTexture(GL_TEXTURE_2D, texture0[i]);
-		// set the texture wrapping parameters
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_BORDER);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_BORDER);
-		// set texture filtering parameters
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
-		// load image, create texture and generate mipmaps
-		int width, height, nrChannels;
-		//stbi_set_flip_vertically_on_load(true); // tell stb_image.h to flip loaded texture's on the y-axis
-		unsigned char *data0 = stbi_load(imgName[i], &width, &height, &nrChannels, 0);
-		data.push_back(data0);
-		if (data[i])
-		{
-			glTexImage2D(GL_TEXTURE_2D, 0, GL_RGB, width, height, 0, GL_RGB, GL_UNSIGNED_BYTE, data[i]);
-			glGenerateMipmap(GL_TEXTURE_2D);
-		}
-		else
-		{
-			std::cout << "Failed to load texture" << std::endl;
-			return false;
-		}
-		stbi_image_free(data[i]);
-	}
-	return true;
-}
-
 /* Render designed pattern on the screen */
 void ScreenData::renderTexture()
-{
+{// TODO: use functions to replace loops
```

Comment:
You can work on this function when you have time. @WanzhuoShi

Reply:
Okay, I will try it tomorrow.